src/clipboard.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use Clipboard;
+use SelectionData;
+use TargetEntry;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi::gpointer;
+use libc::c_uint;
+use std::boxed::Box as Box_;
+use std::mem::transmute;
+
+pub trait ClipboardExtManual {
+    fn set_with_data<F: Fn(&Clipboard, &SelectionData, u32) + 'static>(&self, targets: &[TargetEntry], f: F) -> bool;
+}
+
+impl<O: IsA<Clipboard>> ClipboardExtManual for O {
+    fn set_with_data<F: Fn(&Clipboard, &SelectionData, u32) + 'static>(&self, targets: &[TargetEntry], f: F) -> bool {
+        let stashed_targets: Vec<_> = targets.iter().map(|e| e.to_glib_none()).collect();
+        let mut t = Vec::with_capacity(stashed_targets.len());
+        for stash in &stashed_targets {
+            unsafe {
+                t.push(ffi::GtkTargetEntry {
+                    target: (*stash.0).target,
+                    flags: (*stash.0).flags,
+                    info: (*stash.0).info,
+                });
+            }
+        }
+        let t_ptr: *mut ffi::GtkTargetEntry = t.as_mut_ptr();
+        let f: Box_<Box_<Fn(&Clipboard, &SelectionData, u32) + 'static>> = Box_::new(Box_::new(f));
+        let user_data = Box_::into_raw(f) as *mut _;
+        let success : bool = unsafe { from_glib(ffi::gtk_clipboard_set_with_data(self.to_glib_none().0,
```

Comment:
Is the closure only called from here or might it also be called after the function returned? In case of the former the static bound is not needed, and boxing neither.

Reply:
It won't be called immediately. Only after the function returns.